File: php/src/Models/MessagePrecheckOut.php
<?php

// this file is @generated
declare(strict_types=1);

namespace Svix\Models;

class MessagePrecheckOut implements \JsonSerializable
{
    private array $setFields = [];

    /**
     * @param bool $active whether there are any active endpoint that would get sent such a message
     */
    private function __construct(
        public readonly bool $active,
        array $setFields = [],
    ) {
        $this->setFields = $setFields;
    }

    /**
     * Create an instance of MessagePrecheckOut with required fields.
     */
    public static function create(
        bool $active,
    ): self {
        return new self(
            active: $active,
            setFields: ['active' => true]
        );
    }

    public function jsonSerialize(): mixed
    {
        $data = [
            'active' => $this->active];

        return \Svix\Utils::newStdClassIfArrayIsEmpty($data);
    }

    /**
     * Create an instance from a mixed obj.
     */
    public static function fromMixed(mixed $data): self
    {
        return new self(
            active: \Svix\Utils::deserializeBool($data, 'active', true, 'MessagePrecheckOut')
        );
    }

    /**
     * Create an instance from a json string.
     */
    public static function fromJson(string $json): self
    {
        $data = json_decode(json: $json, associative: true, depth: 512, flags: JSON_THROW_ON_ERROR);

        return self::fromMixed($data);
    }
}

File: java/lib/src/main/java/com/svix/models/MessagePrecheckOut.java
// This file is @generated
package com.svix.models;

import com.fasterxml.jackson.annotation.JsonAutoDetect;
import com.fasterxml.jackson.annotation.JsonAutoDetect.Visibility;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.core.JsonProcessingException;
import com.svix.Utils;

import lombok.EqualsAndHashCode;
import lombok.ToString;

@ToString
@EqualsAndHashCode
@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonAutoDetect(getterVisibility = Visibility.NONE, setterVisibility = Visibility.NONE)
public class MessagePrecheckOut {
    @JsonProperty private Boolean active;

    public MessagePrecheckOut() {}

    public MessagePrecheckOut active(Boolean active) {
        this.active = active;
        return this;
    }

    /**
     * Whether there are any active endpoint that would get sent such a message.
     *
     * @return active
     */
    @javax.annotation.Nonnull
    public Boolean getActive() {
        return active;
    }

    public void setActive(Boolean active) {
        this.active = active;
    }

    /**
     * Create an instance of MessagePrecheckOut given an JSON string
     *
     * @param jsonString JSON string
     * @return An instance of MessagePrecheckOut
     * @throws JsonProcessingException if the JSON string is invalid with respect to
     *     MessagePrecheckOut
     */
    public static MessagePrecheckOut fromJson(String jsonString) throws JsonProcessingException {
        return Utils.getObjectMapper().readValue(jsonString, MessagePrecheckOut.class);
    }

    /**
     * Convert an instance of MessagePrecheckOut to an JSON string
     *
     * @return JSON string
     */
    public String toJson() throws JsonProcessingException {
        return Utils.getObjectMapper().writeValueAsString(this);
    }
}
